modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/processor/DefaultSegmentsExperienceRequestProcessor.java

```diff
@@ -46,17 +46,15 @@ public long[] getSegmentsExperienceIds(
 			long classNameId, long classPK, long[] segmentsExperienceIds)
 		throws PortalException {
 
-		List<SegmentsExperience> segmentsExperiences =
-			_segmentsExperienceLocalService.getSegmentsExperiences(
-				groupId, classNameId, classPK, true);
+		List<SegmentsExperience> segmentsExperiencesAfterFilter =
+			ListUtil.filter(
+				_segmentsExperienceLocalService.getSegmentsExperiences(
+					groupId, classNameId, classPK, true),
+				segmentsExperience -> segmentsExperience.getPriority() >= 0);
 
-		Stream<SegmentsExperience> stream = segmentsExperiences.stream();
-
-		return stream.filter(
-			segmentsExperience -> segmentsExperience.getPriority() >= 0
-		).mapToLong(
-			SegmentsExperienceModel::getSegmentsExperienceId
-		).toArray();
+		return TransformUtil.transformToLongArray(
+			segmentsExperiencesAfterFilter,
```

Comment:
@jeffwu0724 You can inline such variable.

Reply:
@jeffwu0724 Additionally, please consider combining the `filter` with `transformToLongArray`. Please double check all of the cases where `ListUtil.filter` is used together with `TransformUtil.transform...`.